doc/client_design.md

```diff
@@ -0,0 +1,129 @@
+# Design Doc: PaddleCloud Client
+
+A Command Line Interface for PaddlePaddle Cloud
+
+---
+
+# Goals:
+
+Developers using PaddlePadle Cloud can use this command-line client for conviniece of manage cloud Deep-Learning jobs, including:
+
+- Submitting a PaddlePaddle cluster training job
+- List jobs that is currently running.
+- List all job history that have been submited.
+- Fetch logs of the jobs.
+- Download output(model data) of the completed job.
+- View user's quota usages in their spaces.
+
+PaddleCloud Client is written in Go, which simplifies building for different platforms like MacOS/Linux/Windows. The client is a binary program named "paddlecloud".
+
+# Client Configurations
+
+Client configurations stores in a file: `~/.paddle/config` in the below `yaml` form:
+```yaml
+datacenter1:
+  username: user
+  usercert: /path/to/user.pem
+  userkey:  /path/to/user-key.pem
+  endpoint: dc1.paddlepaddle.org
+```
+
+# Client commands
+
+## Reference
+
+- `paddlecloud submit [options] [package path]`: submit job to PaddlePaddle Cloud
+    - `[package path]`: ***Required*** Job package to submit. Including user training programe and it's dependcies.
+    - `-p --parallism`: Number of parrallel trainers. Defaults to 1.
+    - `-c --cpu`: CPU resource each trainer will use. Defaults to 1.
+    - `-g --gpu`: GPU resource each trainer will use. Defaults to 0.
+    - `-m --memory`: Memory resource each trainer will use. Defaults to 1Gi.
+    - `-s --pservers`: Number of parameter servers. Defaults equal to `-p`
+    - `-u --pscpu`: Parameter server CPU resource. Defaults to 1.
+    - `-y --psmemory`: Parameter server momory resource. Defaults to 1Gi.
+    - `-e --entry`: Command of starting trainer process. Defaults to `paddle train`
+    - `-t --topology`: ***Will Be Deprecated*** `.py` file contains paddle v1 job configs
+- `paddlecloud jobs [-a]`: List jobs. List only running jobs if no `-a` specified.
+    - `-a`: List all jobs.
+- `paddlecloud logs [job]`: Print logs of the job.
+- `paddlecloud quota`: Show user's quota usages.
+- `paddlecloud pfs ...`: PaddlePaddle Cloud data management.
+    `<dest>` is the path on the cloud. The form must be like `/pfs/$DATACENTER/home/$USER`. `$DATACENTER` is the from configuration where you setup at `~/.paddle/config`
+    - `paddlecloud pfs cp <local src> [<local src> ... ] <remote dest>`: Upload a file
+    - `paddlecloud pfs cp <remote src> [<remote src> ... ] <local dest>`: Download a file
+    - `paddlecloud pfs ls <remote dir>`: List files under `<remote dir>`.
+
+## Examples
+
+A Sample job package may contain files like below:
+
+```
+job_word_emb/
```

Comment:
I don't think `train.py` and `data` directory in the same level is a good idea. 
At the first step of `paddlecloud submit`, it will collecting the trainer package into a tar file, if data directory in the same level directory, we should `tar -czvf --exculde ./data ...` and the data directory maybe any name, so I think this one is better:
```bash
job_word_emb/
|-- module
    |-- trainer.py
    |-- dict1.pickle
    `-- my_topo.py
`-- data
    |-- train
        |-- train.txt-000
        ...
    `--test
        |-- test.txt-000
        ...
```